lib/index.d.ts

```diff
@@ -39,3 +45,76 @@ export class StateController<State> {
   /** @internal Unsubscribe the listener callback that was passed to subscribeUpdates */
   unsubscribeUpdates(listener: (props: Partial<State>) => void): void;
 }
+
+export class Component extends WebComponent {
```

Comment:
could we make Component a generic i.e `export class Component<State> extends WebComponent`

this makes update, shouldUpdate, state, setState, all typechecked for sanity.